tests/testthat/test-orderly.R

```diff
@@ -1,235 +1,124 @@
+
+  remote_path <- tempfile()
+  cl <- orderly_remote_sharedrive(remote_path)
+
+  expect_equal(cl$list_reports(), character(0))
+
+  expect_error(cl$push("/not/a/path"), "Can't push report at path")
+
+  res <- cl$push(paths$report_paths$example[1])
+
+  expect_equal(cl$list_reports(), "example")
+  expect_length(cl$list_versions("example"), 1)
+
+  example <- file.path(remote_path, "archive", "example")
+  expect_true(dir.exists(example))
+  expect_true(file.exists(file.path(
+    example, basename(paths$report_paths$example[1]))))
+
+  res <- cl$push(paths$report_paths$example[2])
+  expect_equal(cl$list_reports(), "example")
+  expect_length(cl$list_versions("example"), 2)
+  expect_true(file.exists(file.path(
+    remote_path, "archive", "example",
+    basename(paths$report_paths$example[2]))))
+
+  res <- cl$push(paths$report_paths$example2[1])
+
+  expect_equal(cl$list_reports(), c("example", "example2"))
+  expect_length(cl$list_versions("example2"), 1)
+
+  example2 <- file.path(remote_path, "archive", "example2")
+  expect_true(dir.exists(example2))
+  expect_true(file.exists(file.path(example2,
+    basename(paths$report_paths$example2[1]))))
+})
+
+test_that("list_reports", {
+  paths <- setup_orderly()
+
+  cl <- orderly_remote_sharedrive(paths$remote)
+  expect_equal(cl$list_reports(), c("example", "example2"))
 })
 
 
-test_that("list_versions calls folder$files('archive/<name>')", {
-  folder <- list(files = mockery::mock(list(name = c("a", "b", "c"))))
-  cl <- orderly_remote_sharepoint_$new(folder)
-  expect_equal(cl$list_versions("x"), c("a", "b", "c"))
-  mockery::expect_called(folder$files, 1)
-  expect_equal(mockery::mock_args(folder$files)[[1]], list("archive/x"))
+test_that("list_versions", {
+  paths <- setup_orderly()
+
+  cl <- orderly_remote_sharedrive(paths$remote)
+  expect_length(cl$list_versions("example"), 2)
+
+  expect_equal(cl$list_versions("unknown"), character(0))
 })
 
 
 test_that("pull", {
-  path <- orderly1::orderly_example("minimal")
-  id <- orderly1::orderly_run("example", root = path, echo = FALSE)
-  p <- orderly1::orderly_commit(id, root = path)
-  zip <- zip_dir(p)
+  paths <- setup_orderly()
+  cl <- orderly_remote_sharedrive(paths$remote)
 
-  folder <- list(download = mockery::mock(zip))
-  cl <- orderly_remote_sharepoint_$new(folder)
-  res <- cl$pull("example", id)
+  res <- cl$pull("example", basename(paths$report_paths$example[1]))
 
   expect_true(file.exists(res))
   expect_true(file.info(res)$isdir)
-  expect_setequal(dir(res), dir(p))
-
-  mockery::expect_called(folder$download, 1)
-  args <- mockery::mock_args(folder$download)[[1]]
-  expect_equal(args[[1]], file.path("archive/example", id))
-  expect_match(args[[2]], "\\.zip$")
-  expect_equal(normalizePath(dirname(args[[2]]), mustWork = TRUE),
-               normalizePath(tempdir(), mustWork = TRUE))
+  expect_setequal(dir(res), dir(paths$report_paths$example[1]))
+  expect_setequal(openssl::md5(dir(res)),
```

Comment:
Why openssl here, in conjunction with setequal?

Reply:
oh set equal just a copy paste mistake!